GHCN.py

```diff
@@ -0,0 +1,1044 @@
+            filename = '{0:s}/ghcnd_all/{1:s}.dly'.format(folder, format(stdef['id']))
+            kwargs = dict() # combine station and variable attributes
+            kwargs.update(stdef); kwargs.update(vardef.getKWargs())
+            station = DailyStationRecord(filename=filename, **kwargs)
+            station.checkHeader() 
+            self.stationlists[varname].append(station)
+            begin_end_dates=[station.begin_year,station.begin_mon,station.end_year,station.end_mon]
+            count=0
+            for key,fct in station_format[-4:]:
+               stdef[key]=fct(str(begin_end_dates[count]))
+               count+=1
+
+    assert len(self.stationlists[varname]) == ns # make sure we got all (lists should have the same length)
+    
+  def prepareDataset(self, filename=None, folder=None):
+    ''' prepare a PyGeoData dataset for the station data (with all the meta data); 
+        create a NetCDF file for monthly data; also add derived variables          '''
+    if folder is None: folder = '{:s}/'.format(root_folder) # default folder scheme 
+    elif not isinstance(folder,basestring): raise TypeError
+    if filename is None: filename = 'ghcn{:s}_monthly.nc'.format(self.datatype)# default folder scheme 
+    #if filename is None: filename = 'ghcn_monthly.nc'# default folder scheme 
+    elif not isinstance(filename,basestring): raise TypeError
+    # meta data arrays
+    dataset = Dataset(atts=self.atts)
+    # station axis (by ordinal number)
+    stationlist = self.stationlists.values()[0] # just use first list, since meta data is the same
+    assert all([len(stationlist) == len(stnlst) for stnlst in self.stationlists.values()]) # make sure none is missing
+    station = Axis(coord=np.arange(1,len(stationlist)+1, dtype='int'), **varatts['station']) # start at 1
+    # station name
+    namelen = max([len(stn.name) for stn in stationlist])
+#    if  os.path.isfile(ncfile)==False:
+    strarray = np.array([stn.name.ljust(namelen) for stn in stationlist], dtype='|S{:d}'.format(namelen))
+    dataset += Variable(axes=(station,), data=strarray, **varatts['name'])
+    # geo locators (lat/lon/alt)
+    for coord in ('lat','lon','alt'):
+      coordarray = np.array([getattr(stn,coord) for stn in stationlist], dtype='float32') # single precision float
+      dataset += Variable(axes=(station,), data=coordarray, **varatts[coord])
+    # start/end dates (month relative to 1980-01)
+    begin_year=np.zeros(len(stationlist), dtype='int16');
+    begin_mon= np.zeros(len(stationlist), dtype='int16');
+    end_year = np.zeros(len(stationlist), dtype='int16');
+    end_mon=np.zeros(len(stationlist), dtype='int16');
+    datearray=np.empty([2,len(stationlist)], dtype='int16');
+    pntcounter=0
+    print self.variables
+    for pnt in ('begin','end'):
+      s=0#station counter
+      for stations in stationlist:
+       filenamedly = '{0:s}/ghcnd_all/{1:s}.dly'.format(folder,stations.id)
+       # open file

```

Comment:
But debag mode is a good idea. 

发自我的 iPhone

> 在 2016年7月24日，15:10，"Andre R. Erler" <notifications@github.com> 写道：
> 
> In GHCN.py:
> 
> > +    for coord in ('lat','lon','alt'):
> > +      coordarray = np.array([getattr(stn,coord) for stn in stationlist], dtype='float32') # single precision float
> > +      dataset += Variable(axes=(station,), data=coordarray, **varatts[coord])
> > +    # start/end dates (month relative to 1980-01)
> > +    begin_year=np.zeros(len(stationlist), dtype='int16');
> > +    begin_mon= np.zeros(len(stationlist), dtype='int16');
> > +    end_year = np.zeros(len(stationlist), dtype='int16');
> > +    end_mon=np.zeros(len(stationlist), dtype='int16');
> > +    datearray=np.empty([2,len(stationlist)], dtype='int16');
> > +    pntcounter=0
> > +    print self.variables
> > +    for pnt in ('begin','end'):
> > +      s=0#station counter
> > +      for stations in stationlist:
> > +       filenamedly = '{0:s}/ghcnd_all/{1:s}.dly'.format(folder,stations.id)
> > +       # open file
> By having a debug-mode, you could make this change automatically...
> 
> ―
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>